beacon-chain/chaintest/backend/simulated_backend.go

```diff
@@ -262,32 +201,142 @@ func (sb *SimulatedBackend) RunStateTransitionTest(testCase *StateTestCase) erro
 		averageDuration(averageTimesPerTransition),
 	)
 
-	if beaconState.Slot != testCase.Results.Slot {
+	if err := sb.compareTestCase(testCase); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// initializeStateTest sets up the environment by generating all the required objects in order
+// to proceed with the state test.
+func (sb *SimulatedBackend) initializeStateTest(testCase *StateTestCase) (hashOnions [][32]byte,
```

Comment:
Do these need to be methods of the simulated backend? It seems to me they look more like pure functions

Reply:
It uses methods on the simulatedBackend, I think this appropriate